File: MainGame/LuaLib/Assembly-CSharp/Task_SearchTaskProto.lua
---@class Task_SearchTaskProto : System.ValueType
---@field public ProtoCode number
---@field public ProtoEnName string
local m = {}

---@overload fun(): @virtual
---@virtual
---@param isChild boolean
---@return string
function m:ToArray(isChild) end

---@overload fun(buffer:string): @static
---@static
---@param buffer string
---@param isChild boolean
---@return Task_SearchTaskProto
function m.GetProto(buffer, isChild) end

Task_SearchTaskProto = m
return m

File: Server/GameServer/GameServerApp/GameServerApp/Proto/Task_SearchTaskProto.cs
//===================================================
//作    者：边涯  http://www.u3dol.com
//创建时间：2019-11-13 22:42:00
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary>
/// 客户端发送查询任务消息
/// </summary>
public struct Task_SearchTaskProto : IProto
{
    public ushort ProtoCode { get { return 15001; } }
    public string ProtoEnName { get { return "Task_SearchTask"; } }


    public byte[] ToArray(MMO_MemoryStream ms, bool isChild = false)
    {
        ms.SetLength(0);
        if (!isChild)
        {
            ms.WriteUShort(ProtoCode);
        }


        return ms.ToArray();
    }

    public static Task_SearchTaskProto GetProto(MMO_MemoryStream ms, byte[] buffer)
    {
        Task_SearchTaskProto proto = new Task_SearchTaskProto();
        ms.SetLength(0);
        ms.Write(buffer, 0, buffer.Length);
        ms.Position = 0;


        return proto;
    }
}